packages/alfa-rules/src/common/expectation/get-colors.ts

```diff
@@ -150,140 +159,182 @@ function getLayers(
         .computed("background-size")
         .value.equals(style.initial("background-size").value)
     ) {
-      return None;
+      return Err.of(Layer.Error.HasBackgroundSize);
     }
 
     // For each gradient, we extract all color stops into a background layer of
     // their own. As gradients need a start and an end point, there will always
     // be at least two color stops.
-    const stops: Array<RGB<Percentage, Percentage>> = [];
-
-    layers.push({ colors: stops, opacity });
+    const stops: Array<Color> = [];
 
     for (const item of image.image.items) {
       if (item.type === "stop") {
-        const color = resolveColor(item.color, style);
+        const color = Color.resolve(item.color, style);
 
         if (color.isSome()) {
           stops.push(color.get());
         } else {
-          return None;
+          return Err.of(Layer.Error.HasUnresolvableGradientStop);
         }
       }
     }
+
+    layers.push(Layer.of(stops, opacity));
   }
 
-  // If the background layer does not have a lower layer that is fully opaque, we
-  // need to also locate the background layers sitting behind the current layer.
-  // As Alfa does not yet implement a layout system, we have to assume that the
-  // DOM tree will reflect the layout at least to some extent; we therefore
-  // simply use the background layers of the ancestors of the element.
-  //
-  // When we have a layout system in place, we should instead use Picasso
-  // (https://github.com/siteimprove/picasso) for spatially indexing the box
-  // tree in which case the background layers sitting behind the current layer
-  // can be found by issuing a range query for the box of the current element.
   if (
     layers.length > 0 &&
-    layers[0].opacity === 1 &&
-    layers[0].colors.every((color) => color.alpha.value === 1)
+    layers.every(
+      (layer) =>
+        layer.opacity === 1 &&
+        layer.colors.every((color) => color.alpha.value === 1)
+    )
   ) {
-    return Option.of(layers);
+    return Result.of(layers);
   }
 
-  const parent = element
-    .parent({
-      flattened: true,
-    })
-    .filter(Element.isElement);
+  if (isPositioned(device, "absolute", "fixed")(element)) {
+    return Err.of(Layer.Error.HasNonStaticPosition);
+  }
+
+  // If the background layer does not have a lower layer that is fully opaque,
+  // we need to also locate the background layers sitting behind the current
+  // layer.
+
+  if (hasInterposedDescendant(device)(element)) {
+    return Err.of(Layer.Error.HasInterposedDescendant);
+  }
```

Comment:
That comment doesn't seem to match the following code.
Looks like it's comment for the previous step, and that comment from here is missing.

Reply:
Good catch, that was a brain fart 🙈